### PR TITLE
Add auto-formatter (Black) to CI

### PR DESCRIPTION
**Pull request by IanVermes**
Python code now has its formatting and style checked as part of the CI loop.

Code is checked against the expected output of the [Black](https://github.com/psf/black) auto-formatter. All existing code is formatted.

### CODE REVIEW

**Comment by IanVermes** [comment]
All python files are formatted now with Black and pass the CI test.
 - Commit 646916f establishes CI setup to use format checking & 
 - Commit cddbece has the

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (dtip) [inline .travis.yml]: This addition looks accidental - please delete :)

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -78,7 +78,9 @@ you already have `virtualenv` and `Git` installed and ready to go.
 
     $ tox
 
-8. If your contribution is a bug fix or new feature, you should add a test to the existing test suite.
```
Comment (dtip) [inline CONTRIBUTING.rst]: You've deleted this important point about adding tests. I think you meant to add a new item to the list of contribution guidelines

File: .travis.yml (hunk 1 above)
Comment (IanVermes) [reply .travis.yml]: Its added back by the yaml autoformatter. What do you suggest?

**Pull request merged**